time_tracker_api/projects/projects_model.py

```diff
@@ -34,19 +34,52 @@ class ProjectCosmosDBModel(CosmosDBModel):
     def __init__(self, data):
         super(ProjectCosmosDBModel, self).__init__(data)  # pragma: no cover 
 
+    def __contains__(self, item):
+        if type(item) is CustomerCosmosDBModel:
+            return True if item.id == self.customer_id else False
+        else:
+            raise NotImplementedError
+
     def __repr__(self):
         return '<Project %r>' % self.name  # pragma: no cover
 
     def __str___(self):
         return "the project \"%s\"" % self.name  # pragma: no cover
 
 
-def create_dao() -> ProjectDao:
-    repository = CosmosDBRepository.from_definition(container_definition,
-                                                    mapper=ProjectCosmosDBModel)
+class ProjectCosmosDBRepository(CosmosDBRepository):
+    def __init__(self):
+        CosmosDBRepository.__init__(self, container_id=container_definition['id'],
+                                    partition_key_attribute='tenant_id',
+                                    mapper=ProjectCosmosDBModel)
+
 
-    class ProjectCosmosDBDao(APICosmosDBDao, ProjectDao):
-        def __init__(self):
-            CosmosDBDao.__init__(self, repository)
+class ProjectCosmosDBDao(APICosmosDBDao, ProjectDao):
+    def __init__(self, repository):
+        CosmosDBDao.__init__(self, repository)
+
+    def get_all(self, conditions: dict = None, **kwargs) -> list:
+        """
+        Get all the projects an active client has
+        :param (dict) conditions: Conditions for querying the database
+        :param (dict) kwargs: Pass arguments
+        :return (list): ProjectCosmosDBModel object list
+        """
+        event_ctx = self.create_event_context("read-many")
+        customer_dao = customers_create_dao()
+        customers = customer_dao.get_all()
+        projects = self.repository.find_all(event_ctx, **kwargs)
```

Comment:
I think you should change this a little bit. 
1. Query all customers (DONE and OK)
2. Query projects `where customer_id IN ( customer_id from the above-query executed)`

With the IN constraint, you will have only projects that belong to customers that are not deleted.

Reply:
So you say, do a query for each customer to find the projects?
I think it is a good solution, but we have to keep in mind the response time for each query, if we have many customers, this solution may be a little slow.

Reply:
Not at all, one query with an IN clause to get all the projects

Reply:
Got it